routes/signup.js

```diff
@@ -0,0 +1,63 @@
+module.exports = (function() {
+  var express = require('express');
+  var router = express.Router();
+
+//Sign in Page
+  router.get('/', function(req, res){
+    res.render('signup', {layout: false});
+  })
+
+//Create Session
+  router.post('/create_user', function(req, res, next) {
+  	var check_query = 'SELECT * FROM users WHERE user_name = ? OR email = ?';
+  	var check_param = [req.body.username, req.body.email];
+	var create_query = 'INSERT INTO users (first_name, last_name, email, user_name, password) VALUES (?, ?, ?, ?, ?)';
+	var create_param = [req.body.firstname, req.body.lastname, req.body.email, req.body.username, req.body.password];
+	var new_id_query = 'SELECT * FROM users WHERE user_name = ?';
+	var new_id_param = [req.body.username];
+	var mysql = req.app.get('mysql');
+
+    new Promise((resolve, reject) => {
+      // check if user already exists in database
+      mysql.pool.query(check_query, check_param, (err, results, fields) => {
```

Comment:
You should also check for database error here before proceeding.

Reply:
Fixed as specified